sbat.c

```diff
@@ -330,7 +333,55 @@ set_sbat_uefi_variable(void)
 	UINT32 attributes = 0;
 
 	UINT8 *sbat = NULL;
+	UINT8 *sbat_policy = NULL;
 	UINTN sbatsize = 0;
+	UINTN sbat_policysize = 0;
+
+	char *sbat_var = NULL;
+	UINTN reset_sbat = 0;
+
+        efi_status = get_variable_attr(SBAT_POLICY, &sbat_policy,
+				       &sbat_policysize,
+                                       SHIM_LOCK_GUID, &attributes);
+	if (EFI_ERROR(efi_status)) {
+		dprint("Default sbat policy: previous\n");
+		sbat_var = SBAT_VAR_PREVIOUS;
+	} else {
+		efi_status = get_variable_attr(SBAT_POLICY, &sbat_policy,
+				               &sbat_policysize,
+					       SHIM_LOCK_GUID, &attributes);
+		if (EFI_ERROR(efi_status)) {
+			dprint("Default sbat policy: previous\n");
+			sbat_var = SBAT_VAR_PREVIOUS;
+		} else {
+			switch (*sbat_policy) {
+				case 1:
+					dprint("Custom sbat policy: latest\n");
+					sbat_var = SBAT_VAR_LATEST;
+					break;
+				case 2:
+					dprint("Custom sbat policy: previous\n");
+					sbat_var = SBAT_VAR_PREVIOUS;
+					break;
+				case 3:
+					if (secure_mode()) {
+						console_print(L"Custom sbat policy: "
+							L"reset refused"
```

Comment:
By the time we get here, it's at seven levels of indentation.  Maybe another function to handle policy reset?

Or tighter use of blocks - maybe something like:

```C
if (EFI_ERROR(...)) {
    dprint("Default sbat policy: previous\n");
    sbat_var = SBAT_VAR_PREVIOUS;
} else if (*sbat_policy == 1) {
    ...
} else if (*sbat_policy == 2) {
    ...
} else if (secure_mode()) {
    ...
} else {
   ...
}
```

?

Reply:
Getting rid of the redundant get_variable_attr() pulled this all out far enough again that I'm no longer bothered by it. Let me know if you want me to refactor it a bit more still, and I'll be happy to do that.